src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/toolbar_js.html.twig

```diff
@@ -455,60 +457,32 @@
 
                 showToolbar: function(token) {
                     var sfwdt = this.getSfwdt(token);
-                    removeClass(sfwdt, 'sf-display-none');
 
-                    if (getPreference('toolbar/displayState') == 'none') {
-                        document.getElementById('sfToolbarMainContent-' + token).style.display = 'none';
-                        document.getElementById('sfToolbarClearer-' + token).style.display = 'none';
-                        document.getElementById('sfMiniToolbar-' + token).style.display = 'block';
+                    if ('closed' === getPreference('toolbar/displayState')) {
+                        sfwdt.classList.add('sf-toolbar-closed');
+                        sfwdt.classList.remove('sf-toolbar-opened');
                     } else {
-                        document.getElementById('sfToolbarMainContent-' + token).style.display = 'block';
-                        document.getElementById('sfToolbarClearer-' + token).style.display = 'block';
-                        document.getElementById('sfMiniToolbar-' + token).style.display = 'none';
+                        sfwdt.classList.add('sf-toolbar-opened');
+                        sfwdt.classList.remove('sf-toolbar-closed');
                     }
                 },
 
                 hideToolbar: function(token) {
                     var sfwdt = this.getSfwdt(token);
-                    addClass(sfwdt, 'sf-display-none');
+                    sfwdt.classList.add('sf-toolbar-closed');
+                    sfwdt.classList.remove('sf-toolbar-opened');
```

Comment:
hiding the whole toolbar is not the same thing than closing it (which still has a visible toggle button)